miio/viomivacuum.py

```diff
@@ -422,22 +717,228 @@ def set_dnd(
+            msg = (
+                "Fake schedule not found. "
+                "Please create a scheduled cleanup with the "
+                "following properties:\n"
+                "* Hour: 00\n"
+                "* Minute: 00\n"
+                "* Select all (minus one) the rooms one by one\n"
+                "* Set as inactive scheduled cleanup\n"
+                "Then create a scheduled cleanup with the room missed at "
+                "previous step with the following properties:\n"
+                "* Hour: 00\n"
+                "* Minute: 00\n"
+                "* Select only the missed room\n"
+                "* Set as inactive scheduled cleanup\n"
+            )
+            raise ViomiVacuumException(msg)
+
+        self._cache["rooms"] = rooms
+        return rooms
+
+    # MISSING Area editor
+
+    # MISSING Reset map
+
+    # MISSING Device leveling
+
+    # MISSING Looking for the vacuum-mop
+
+    @command()
+    def consumable_status(self) -> ViomiConsumableStatus:
+        """Return information about consumables."""
+        return ViomiConsumableStatus(self.send("get_consumables"))
+
+    @command(
+        click.argument("direction", type=EnumType(ViomiMovementDirection)),
+        click.option(
+            "--duration",
+            type=float,
+            default=0.5,
+            help="number of seconds to perform this movement",
+        ),
+    )
+    def move(self, direction: ViomiMovementDirection, duration=0.5):
+        """Manual movement."""
+        start = time.time()
+        while time.time() - start < duration:
+            self.send("set_direction", [direction.value])
+            time.sleep(0.1)
+        self.send("set_direction", [ViomiMovementDirection.Stop.value])
+
     @command(click.argument("language", type=EnumType(ViomiLanguage)))
     def set_language(self, language: ViomiLanguage):
-        """Set the device's audio language."""
+        """Set the device's audio language.
+
+        This seems doing nothing on STYJ02YM
+        """
         return self.send("set_language", [language.value])
 
     @command(click.argument("state", type=EnumType(ViomiLedState)))
     def led(self, state: ViomiLedState):
-        """Switch the button leds on or off."""
+        """Switch the button leds on or off.
+
+        This seems doing nothing on STYJ02YM
+        """
         return self.send("set_light", [state.value])
 
     @command(click.argument("mode", type=EnumType(ViomiCarpetTurbo)))
     def carpet_mode(self, mode: ViomiCarpetTurbo):
-        """Set the carpet mode."""
+        """Set the carpet mode.
+
+        This seems doing nothing on STYJ02YM
+        """
         return self.send("set_carpetturbo", [mode.value])
 
-    @command()
-    def fan_speed_presets(self) -> Dict[str, int]:
-        """Return dictionary containing supported fanspeeds."""
-        return {x.name: x.value for x in list(ViomiVacuumSpeed)}
+    @classmethod
+    def get_device_group(cls):
```

Comment:
We don't want to have code duplication for this "hack" from mirobo, rather, we want to have this on the miiocli level for all supported devices. I wrote a bit about the plan here https://github.com/rytilahti/python-miio/issues/728#issuecomment-685967596 .

Reply:
ho wow, this is a huge change. I'm removing this change for now